models/classes/preview/ItemPreviewerService.php

```diff
@@ -119,4 +138,60 @@ public function unregisterAdapter($moduleId)
         }
         return false;
     }
+
+    /**
+     * Registers a previewer plugin
+     * @param DynamicModule $module the plugin to register
+     * @return boolean true if registered
+     */
+    public function registerPlugin(DynamicModule $module)
+    {
+        if (!is_null($module) && !empty($module->getModule())) {
+            $registry = $this->getRegistry();
+            $config = [];
+            if ($registry->isRegistered(self::REGISTRY_ENTRY_KEY)) {
+                $config = $registry->get(self::REGISTRY_ENTRY_KEY);
+            }
+
+            $index = false;
+            if (isset($config[self::PLUGINS_KEY])) {
+                $index = array_search($module->getModule(), array_column($config[self::PLUGINS_KEY], 'module'));
+            }
+            if ($index === false) {
+                $config[self::PLUGINS_KEY][] = $module->toArray();
+            } else {
+                $config[self::PLUGINS_KEY][$index] = $module->toArray();
+            }
+            $registry->set(self::REGISTRY_ENTRY_KEY, $config);
+            return true;
+        }
+        return false;
```

Comment:
Consider applying an early return in the exceptional case instead of wrapping the whole useful part into a conditional.
```suggestion
        if (null === $module || empty($module->getModule())) {
            return false;
        }

        $registry = $this->getRegistry();
        $config = [];
        if ($registry->isRegistered(self::REGISTRY_ENTRY_KEY)) {
            $config = $registry->get(self::REGISTRY_ENTRY_KEY);
        }

        $index = false;
        if (isset($config[self::PLUGINS_KEY])) {
            $index = array_search($module->getModule(), array_column($config[self::PLUGINS_KEY], 'module'));
        }
        if ($index === false) {
            $config[self::PLUGINS_KEY][] = $module->toArray();
        } else {
            $config[self::PLUGINS_KEY][$index] = $module->toArray();
        }
        $registry->set(self::REGISTRY_ENTRY_KEY, $config);

        return true;
```